### ISSUE TITLE
Task List for DS2 on Paddle

### ISSUE THREAD

**Issue by xinghai-sun** [PaddlePaddle/models#176]
For tracking progress of key tasks:

- [x] Warp-CTC‘s numerical instability problem and INF bug.
- [ ] Bad convergence for THCHS-30.
- [x] Performance drop for latest paddle version.
- [x] Inconsistent results between trainer.test and inference.infer ([issue](https://github.com/PaddlePaddle/models/issues/178)).
- [x] Prefix-beam-search decoder's difference with speech_dl codes.
- [ ] Double check for the reliability of Paddle-v2 (by comparing V1 and V2).
- [x] Experimenting with MFCC.
- [x] Kubernetes running and profiling.
- [ ] Performance profiling.
- [x] Make autoset for cudnn-conv and cudnn-batch-norm.
- [x] Speed difference for gpu-00 and gpu-01 machines.
- [x] Debug for incorrect inferencing with multiple GPUs ([issue](https://github.com/PaddlePaddle/Paddle/issues/3073)).
- [x] Unicom-3k data preprocessing.
- [ ] Remove padding steps from ctc loss computation.
- [x] Fix CUDNN_STATUS_NOT_SUPPORTED error ([Issue](https://github.com/PaddlePaddle/Paddle/issues/929)).
- [x] Make setup.sh more robust. #177 #182 
- [x] Add a real-time ASR demo (server + client) ([PR](https://github.com/PaddlePaddle/models/pull/186)).

https://github.com/PaddlePaddle/Paddle/projects/17

**Comment by lcy-seso** [comment]
@xinghai-sun autoset cudnn-conv and cudnn-batchnorm is fixed in this PR https://github.com/PaddlePaddle/Paddle/pull/3049/files

**Comment by xinghai-sun** [comment]
@lcy-seso Great! Thanks!

**Comment by xinghai-sun** [comment]
[Performance drop for latest paddle version] Done.

After this [issue](https://github.com/PaddlePaddle/Paddle/issues/3042) fixed, the training convergence rate of cudnn_conv and cudnn_batch_norm is consistent with that of exconv and batch_norm.

However, when doing inferencing we run into this CUDNN_STATUS_NOT_SUPPORTED error ([Issue](https://github.com/PaddlePaddle/Paddle/issues/929)).

**Comment by xinghai-sun** [comment]
Inconsistent results between trainer.test and inference.infer ([issue](https://github.com/PaddlePaddle/models/issues/178)). 

Done.

**Comment by pkuyym** [comment]
Warp-CTC's numerical instability and INF bug are fixed by [#3105](https://github.com/PaddlePaddle/Paddle/pull/3105) and  [#5](https://github.com/gangliao/warp-ctc/pull/5), details are in [#3076](https://github.com/PaddlePaddle/Paddle/issues/3076).

**Comment by kuke** [comment]
Seems that ```cuDNN >= 6.0```  has fixed the CUDNN_STATUS_NOT_SUPPORTED error we meet. Update [README in DS2](https://github.com/PaddlePaddle/models/pull/185) and add [cuDNN version check](https://github.com/PaddlePaddle/Paddle/pull/3185) in Paddle accordingly.

**Comment by shanyi15** [comment]
您好，此issue在近一个月内暂无更新，我们将于今天内关闭。若在关闭后您仍需跟进提问，可重新开启此问题，我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解~感谢您对PaddlePaddle的支持! 
 Hello, this issue has not been updated in the past month. We will close it today for the sake of other user‘s experience. If you still need to follow up on this question after closing, please feel free to reopen it. In that case, we will get back to you within 24 hours. We apologize for the inconvenience caused by the closure and thank you so much for your support of PaddlePaddle Group!
